src/model_execution_worker/tasks.py

```diff
@@ -285,3 +347,31 @@ def on_error(request, ex, traceback, record_task_name, analysis_pk, initiator_pk
         args=(analysis_pk, initiator_pk, traceback),
         queue='celery'
     ).delay()
+
+
+def prepare_complex_model_file_inputs(complex_model_files, upload_directory, run_directory):
+    """Places the specified complex model files to the run_directory.
+
+    The unique upload filenames are converted back to the original upload names, so that the
+    names match any input configuration file.
+
+    On Linux, the files are symlinked, whereas on Windows the files are simply copied.
+
+    Args:
+        complex_model_files (list of complex_model_data_file): List of namedtuples giving the files
+            to make available.
+        upload_directory (str): Source directory containing the uploaded files with unique filenames.
+        run_directory (str): Model inputs directory to place the files in.
+
+    Returns:
+        None.
+
+    """
+    for cmf in complex_model_files:
+        from_path = os.path.join(upload_directory, cmf["stored_filename"])
```

Comment:
Aha, good spot - any idea where you spotted that comment? It was once a namedtuple, but it turns out that because the data gets converted to JSON and back, it started turning up as a dict in some places and a namedtuple in others, which got very confusing.